server/channels/app/export.go

```diff
@@ -831,16 +833,52 @@ func (a *App) ListExports() ([]string, *model.AppError) {
 	return results, nil
 }
 
+func (a *App) GeneratePresignURLForExport(name string) (*model.PresignURLResponse, *model.AppError) {
+	if !a.Config().FeatureFlags.EnableExportDirectDownload {
+		return nil, model.NewAppError("GeneratePresignURLForExport", "app.eport.generate_presigned_url.featureflag.app_error", nil, "", http.StatusInternalServerError)
+	}
+
+	if !*a.Config().FileSettings.DedicatedExportStore {
+		return nil, model.NewAppError("GeneratePresignURLForExport", "app.eport.generate_presigned_url.config.app_error", nil, "", http.StatusInternalServerError)
+	}
+
+	b := a.ExportFileBackend()
+	backend, ok := b.(filestore.FileBackendWithLinkGenerator)
+	if !ok {
+		return nil, model.NewAppError("GeneratePresignURLForExport", "app.eport.generate_presigned_url.driver.app_error", nil, "", http.StatusInternalServerError)
+	}
+
+	p := path.Join(*a.Config().ExportSettings.Directory, "/", name)
```

Comment:
`path.Join` adds slashes automatically.

```suggestion
	p := path.Join(*a.Config().ExportSettings.Directory, name)
```